src/reporter.ts

```diff
@@ -0,0 +1,290 @@
+    curr.status = suiteStatus as SauceStatus;
+    curr.metadata = {
+      attempt: parsed.testCase?.attempt,
+      sourceLocation: parsed.testCase?.sourceLocation,
+    }
+    this.consoleLog.push(`${curr.name}\t#${suite.name}`)
+
+    parsed.testSteps.forEach((testStep) => {
+      this.consoleLog.push('  ' + testStep.keyword + (testStep.text || '') + ' - ' + Status[testStep.result.status]);
+      const test = new Test(`${testStep.keyword}${testStep.text || ''}`)
+      const testStatus = testStep.result?.status?.toLowerCase();
+      test.status = testStatus as SauceStatus;
+      test.output = testStep.result?.message;
+      test.duration = this.getDuration(testStep.result?.duration);
+      test.attachments = [];
+      testStep.attachments.forEach((attachment: any) => {
+        this.assets.push({
+          filename: attachment.testCaseStartedId + '.log',
+          data: Buffer.from(attachment.body),
+        })
+        test.attachments?.push({
+          name: attachment.testCaseStartedId + '.log',
+          contentType: attachment.mediaType,
+          path: ''
+        })
+      })
+      curr.addTest(test);
+    })
+    suite.addSuite(curr)
+    this.testRun.addSuite(suite);
+  }
+
+  async logTestRun(testRunFinished: { success: any; }) {
+    this.passed = testRunFinished.success;
+    this.reportToFile(this.testRun)
+
+    const id = await this.reportToSauce(this.testRun);
+    this.consoleLog.push(testRunFinished.success ? 'SUCCESS' : 'FAILURE')
+    this.logSauceJob(id as string)
+    this.log('\n')
+  }
+
+  getDuration(duration: any) {
+    return (duration.seconds * 1000) + (duration.nanos / 1000000000);
+  }
+
+  logSauceJob (jobId: string) {
+    if (jobId.length < 1) {
+      let msg = '';
+      const hasCredentials = process.env.SAUCE_USERNAME && process.env.SAUCE_USERNAME !== '' && process.env.SAUCE_ACCESS_KEY && process.env.SAUCE_ACCESS_KEY !== '';
+      if (!hasCredentials) {
+        msg = `\nNo results reported to Sauce. $SAUCE_USERNAME and $SAUCE_ACCESS_KEY environment variables must be defined in order for reports to be uploaded to Sauce.`;
+      }
+      this.log(msg);
+      return;
+    }
+    const jobUrl = this.getJobUrl(jobId as string, this.region)
+    this.log(`\nReported jobs to Sauce Labs:\n`);
+    this.log(jobUrl)
+  }
+
+  reportToFile(report: TestRun) {
+    if (!this.outputFile) {
+      return;
+    }
+
+    fs.mkdirSync(path.dirname(this.outputFile), { recursive: true });
+    report.toFile(this.outputFile);
+  }
+
+  async reportToSauce(report: TestRun) : Promise<string | undefined> {
+    // Currently no reliable way to get the browser version
+    const browserVersion = '1.0';
+
+    const jobBody = this.createBody({
+      build: this.build,
+      startedAt: this.startedAt ? this.startedAt.toISOString() : new Date().toISOString(),
+      endedAt: this.endedAt ? this.endedAt.toISOString() : new Date().toISOString(),
```

Comment:
startedAt and endedAt are never set so the job duration in the ui will always be 0s.